### PR TITLE
Add redirect test

### LINKED ISSUE #32: Add Redirect tests
https://github.com/Kuadrant/authorino/blob/main/docs/user-guides/deny-with-redirect-to-login.md

Authority can deny unauthorized requests and then redirect them to a different page

### PR DESCRIPTION
**Pull request by JaurbanRH**
Closes #32

### CODE REVIEW

File: testsuite/openshift/objects/auth_config.py [hunk 1]
```diff
@@ -83,8 +83,8 @@ def add_api_key_identity(self, name, all_namespaces: bool = False,
         if match_label:
             matcher.update({
                 "matchLabels": {
-                        "group": match_label
-                    }
+                    "group": match_label
```
Comment (pehala) [inline testsuite/openshift/objects/auth_config.py:86]: Unnecessary change

File: testsuite/tests/kuadrant/authorino/identity/test_redirect.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+from testsuite.openshift.objects.auth_config import AuthConfig
+
+STATUS_CODE = 302
+REDIRECT_URL = 'http://anything.inavlid?redirect_to='
+
+
+@pytest.fixture(scope="module")
+def authorization(authorization, authorino, envoy, blame, openshift, module_label, oidc_provider):
+    """In case of Authorino, AuthConfig used for authorization"""
+    if authorization is None:
```
Comment (pehala) [inline testsuite/tests/kuadrant/authorino/identity/test_redirect.py:16]: Should never happen

File: testsuite/tests/kuadrant/authorino/identity/test_redirect.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        authorization = AuthConfig.create_instance(openshift, blame("ac"),
+                                                   envoy.hostname, labels={"testRun": module_label})
+    authorization.add_oidc_identity("rhsso", oidc_provider.well_known["issuer"])
```
Comment (pehala) [inline testsuite/tests/kuadrant/authorino/identity/test_redirect.py:19]: I would move this test to `authorino/` dir only and in that case this will be redundant

File: testsuite/openshift/objects/auth_config.py [hunk 4]
```diff
@@ -156,3 +156,9 @@ def add_response(self, response):
         """Adds response section to authconfig."""
         responses = self.model.spec.setdefault("response", [])
         responses.append(response)
+
+    @modify
+    def add_deny_with(self, code, value):
```
Comment (pehala) [inline testsuite/openshift/objects/auth_config.py:161]: Since you cannot add multiple deny with, it should be called `set_deny_with`

Comment (pehala) [inline testsuite/openshift/objects/auth_config.py:161]: Add this to the interface as well

**Review by pehala — APPROVED** [review]

**Review by jsmolar — APPROVED** [review]
